### ISSUE TITLE
R.Swift grinds Xcode to a hault when file is selected

### ISSUE THREAD

**Issue by philosopherdog** [mac-cain13/R.swift#725]
We're using R.Swift for localized strings. When I select the file Xcode beach balls and becomes unresponsive for like 5 minutes. I need to solve this issue.

**Comment by philosopherdog** [comment]
Incidently the file is 28_000+ lines.

**Comment by ndemie** [comment]
I'm running in to the same issue

**Comment by nashfive** [comment]
Not sure if this can be "solved" by code. I mean, yeah reducing the number of lines would help, but if you have a lot of translations, then it's impossible to reduce the size of the file. My generated file is "only" 718Ko, 13'000+ lines and it's already slowing/freezing Xcode during at least a minute...

**Comment by philosopherdog** [comment]
Yeah this is a pretty serious problem in this project. Rswift is a hard dependency that's everywhere in this code. I am going to have to figure out a way to remove it from strings files as I'm pretty sure this is the problem. Anyhow, it doesn't seem like there's any easy solution. FYI, this is happening on both Apple Silicone and Intel Macs.

**Comment by philosopherdog** [comment]
Hmm. Actually I'm noticing with Xcode 13.2.1 I can select the file again without any issues.

**Comment by gwk** [comment]
I'm working on a project where the generated file is 36k+ lines long. It is creating rather serious usability problems. One minor improvement would be to split the substructures into different generated files, e.g. `R.segue`, `R.storyboard`, `R.color`, etc. This would still leave us with a 30k `R.strings.generated.swift` (or whatever the naming convention would be), but it would at least make visiting the other types less of a disaster.

Another step might to split the strings up into multiple files, but I'm not sure.

**Comment by nicolasvergoz** [comment]
I'm agree with @gwk about splitting the files. Maybe as a boolean option false by default
